client/main.lua

```diff
@@ -1,274 +1,223 @@
+        if not ispolice or isems then return lib.disableRadial(true) end
+        lib.clearRadialItems()
+        lib.addRadialItem({
+            id = 'emergencybutton2',
+            label = Lang:t("options.emergency_button"),
+            icon = 'circle-exclamation',
+            onSelect = function (_, _)
+                if ispolice then
+                    TriggerEvent('police:client:SendPoliceEmergencyAlert')
+                elseif isems then
+                    TriggerEvent('ambulance:client:SendAmbulanceEmergencyAlert')
+                end
+                lib.hideRadial()
+            end
+        })
     else
-        TriggerEvent('qb-radialmenu:client:onRadialmenuClose')
+        lib.clearRadialItems()
+        SetupRadialMenu()
+        lib.disableRadial(false)
     end
+end)
 
-    SetNuiFocus(bool, bool)
-    if sendMessage then
-        SendNUIMessage({
-            action = "ui",
-            radial = bool,
-            items = FinalMenuItems
-        })
+RegisterNetEvent('radialmenu:client:ChangeSeat', function(id, label)
+    local Veh = GetVehiclePedIsIn(cache.ped, false)
+    local IsSeatFree = IsVehicleSeatFree(Veh, id - 2)
+    local speed = GetEntitySpeed(Veh)
+    local HasHarness = exports['qb-smallresources']:HasHarness()
+    if HasHarness then
+        return QBCore.Functions.Notify(Lang:t("error.race_harness_on"), 'error')
     end
-    if delay then Wait(500) end
-    inRadialMenu = bool
-end
-
--- Command
 
-RegisterCommand('radialmenu', function()
-    if ((IsDowned() and IsPoliceOrEMS()) or not IsDowned()) and not PlayerData.metadata["ishandcuffed"] and not IsPauseMenuActive() and not inRadialMenu then
-        setRadialState(true, true)
-        SetCursorLocation(0.5, 0.5)
+    if not IsSeatFree then
+       return QBCore.Functions.Notify(Lang:t("error.seat_occupied"), 'error')
     end
+    
+    local kmh = speed * 3.6
+    
+    if kmh > 100.0 then
+       return QBCore.Functions.Notify(Lang:t("error.vehicle_driving_fast"), 'error')
+    end
+    
+    SetPedIntoVehicle(cache.ped, Veh, id - 2)
+    QBCore.Functions.Notify(Lang:t("info.switched_seats", {seat = label}))
 end)
 
-RegisterKeyMapping('radialmenu', Lang:t("general.command_description"), 'keyboard', 'F1')
-
--- Events
-
--- Sets the metadata when the player spawns
-RegisterNetEvent('QBCore:Client:OnPlayerLoaded', function()
-    PlayerData = QBCore.Functions.GetPlayerData()
-end)
-
--- Sets the playerdata to an empty table when the player has quit or did /logout
-RegisterNetEvent('QBCore:Client:OnPlayerUnload', function()
-    PlayerData = {}
-end)
-
--- This will update all the PlayerData that doesn't get updated with a specific event other than this like the metadata
-RegisterNetEvent('QBCore:Player:SetPlayerData', function(val)
-    PlayerData = val
+RegisterNetEvent('qb-radialmenu:trunk:client:Door', function(plate, door, open)
+    local veh = GetVehiclePedIsIn(cache.ped)
+    if veh == 0 then return end
```

Comment:
```suggestion
    local veh = cache.vehicle
    if not veh then return end
```

Same with this one we utilize cache as much as possible

Reply:
Just confirming, does cache.vehicle return nil or 0 if you are in no vehicle?

Because if it returns 0 like GetVehiclePedIsIn, the if statement underneath will never activate.

Reply:
> 

It returns false